bnd-maven-plugin-parent/bnd-maven-plugin/src/main/java/org/bndtools/maven/BndMavenPlugin.java

```diff
@@ -0,0 +1,176 @@
+package org.bndtools.maven;
+
+/*
+ * Copyright 2001-2005 The Apache Software Foundation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import static aQute.lib.io.IO.getFile;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.project.MavenProject;
+
+import aQute.bnd.build.Project;
+import aQute.bnd.osgi.Builder;
+import aQute.bnd.osgi.Constants;
+import aQute.bnd.osgi.FileResource;
+import aQute.bnd.osgi.Jar;
+import aQute.bnd.osgi.Resource;
+import aQute.lib.io.IO;
+
+@Mojo(name = "bnd-process", defaultPhase = LifecyclePhase.PROCESS_CLASSES, requiresDependencyResolution = ResolutionScope.COMPILE)
+public class BndMavenPlugin extends AbstractMojo {
+	
+	@Parameter(defaultValue = "${project.build.directory}", readonly = true)
+	private File targetDir;
+	
+	@Parameter(defaultValue = "${project.build.outputDirectory}", readonly = true)
+	private File classesDir;
+	
+	@Parameter(defaultValue = "${project.build.outputDirectory}/META-INF/MANIFEST.MF", readonly = true)
+	private File manifestPath;
+	
+	@Parameter(defaultValue = "${project}", required = true, readonly = true)
+	private MavenProject project;
+	
+	@Parameter(property = "bnd.trace", defaultValue = "false", readonly = true)
+	private boolean trace;
+
+	public void execute() throws MojoExecutionException {
+		File bndFile = new File(project.getBasedir(), Project.BNDFILE);
+		
+		Builder builder = new Builder();
+		builder.setTrace(trace);
```

Comment:
In the gradle plugin, I set the trace value based upon the current logging level. See <https://github.com/bndtools/bnd/blob/master/biz.aQute.bnd.gradle/src/aQute/bnd/gradle/BundleTaskConvention.groovy#L102>. Perhaps the same could be done for this to avoid the need to a separate option?